commit ecf039a28e62e689dd0cb61257627fadc82dce00
Author: Matthew Booth <mbooth@redhat.com>
Date:   Mon Oct 22 13:57:35 2012 +0100

    RedHat: Clean RPM database before any rpm operations
    
    We were previously only cleaning the RPM database before doing anything which
    modified the rpm DB. However, I hit a situation where an rpm -q in _init_kernels
    was hanging due to stale state files.
    (cherry picked from commit 9fb11799546d177f8e52a5cae7fa3ac205aebc3a)

diff --git a/lib/Sys/VirtConvert/Converter/RedHat.pm b/lib/Sys/VirtConvert/Converter/RedHat.pm
index 9876a96..7c034bb 100644
--- a/lib/Sys/VirtConvert/Converter/RedHat.pm
+++ b/lib/Sys/VirtConvert/Converter/RedHat.pm
@@ -118,6 +118,8 @@ sub convert
     croak("convert called without meta argument") unless defined($meta);
     croak("convert called without options argument") unless defined($options);
 
+    _clean_rpmdb($g);
+
     _init_grub($g, $root, $desc);
     my $grub_conf = $desc->{boot}->{grub_conf};
 
@@ -125,8 +127,6 @@ sub convert
     _init_augeas($g, $grub_conf);
     _init_kernels($g, $desc);
 
-    _clean_rpmdb($g);
-
     # Un-configure HV specific attributes which don't require a direct
     # replacement
     _unconfigure_hv($g, $root, $desc);
